### DISCUSSION TITLE
S3 uploads work, but files don't load afterwards

### DISCUSSION THREAD

**Discussion by kerberjg** [outline/outline#7587] [Self Hosting]
Since a few days all uploads to Scaleway Object Storage create objects with (object-level!) access set to private.

I have AWS_S3_ACL set to `public-read`, and made no changes to Outline configuration (was running 0.74, now upgraded to 0.79.1, same behavior).

**To Reproduce**
(with S3 set up previously)
1. Go to an existing document
2. Upload an image

What happens:
- Loading bar completes
- The image stays translucent (can't delete it, disappears upon reload)
- Upload succeeded (file present in bucket)
- File GET from bucket fails with 403
- File access is set to "private"

Changing the file manually in the bucket to "public" makes it so it can be fetched at the link used by outline, but since the GET fails the image isn't saved in the document upon reload

**Expected behavior**
Image uploads, and shows up in document.

**Outline (please complete the following information):**
- Install: self hosted
- Version: 0.79.1

**Desktop (please complete the following information):**
 - OS: macOS 14.6
 - Browser: Chrome 128.0.6613.120

**Comment by kerberjg** [comment]
cc @marcociarrocca

**Comment by tommoor** [comment]
I don't know anything about Scaleway Object Store, if this was working on 0.74.0 and then suddenly stopped working I'd suggest your problem is with Scaleway and not Outline specifically. Try and debug what the difference is in files and their permissions uploaded previously vs today.

**Reply by kerberjg** [reply]
I figured it might be a Scaleway issue, thought I'd open an issue anyway if more people are having this issue.

As mentioned above, old files have an ACL of "public" and the new ones have "private", no other options ("public-read") seem available, which is weird

**Reply by guedressel** [reply]
I've got the same Issue with our Scaleway Object Storage bucket.

Following policy allows now public access to all object in `public/` folder

```
{
  "Version": "2023-04-17",
  "Id": "...",
  "Statement": [
    ...,
    {
      "Sid": "public-access",
      "Effect": "Allow",
      "Principal": "*",
      "Action": "s3:GetObject",
      "Resource": "<bucket>/public/*"
    },
    ...
  ]
}
```

I hope this policy is correct and does not pose any too loose permission for our Outline-bucket.  
Looking forward to getting feedback from you, if I put the right policy in place.
